tests/apiserver-proxy-tests/fetch-token/internal/idtokenprovider.go

```diff
@@ -61,45 +109,56 @@ func (p *dexIdTokenProvider) implicitFlow() (map[string]string, error) {
 	case http.StatusFound:
 	case http.StatusOK:
 	default:
-		return nil, fmt.Errorf("got unexpected response on authorize: %d - %s", authorizeResp.StatusCode, authorizeRespBody)
+		return "", fmt.Errorf("got unexpected response on authorize: %d - %s", authorizeResp.StatusCode, authorizeRespBody)
 	}
 
 	var loginEndpoint string
 	if authorizeResp.StatusCode == http.StatusOK {
 		b := bytes.NewBufferString(authorizeRespBody)
 		loginEndpoint, err = getLocalAuthEndpoint(b)
 		if err != nil {
-			return nil, errors.Wrapf(err, "while fetching link to static authentication")
+			return "", errors.Wrapf(err, "while fetching link to static authentication")
 		}
 	} else {
 		loginEndpoint = authorizeResp.Header.Get("location")
 		if strings.Contains(loginEndpoint, "#.*error") {
-			return nil, fmt.Errorf("login - Redirected with error: '%s'", loginEndpoint)
+			return "", fmt.Errorf("login - Redirected with error: '%s'", loginEndpoint)
 		}
 	}
 
+	return loginEndpoint, nil
+}
+
+func (p *dexIdTokenProvider) loginCall(loginEndpoint string) (string, error) {
+
 	if _, err := p.httpClient.Get(p.config.DexConfig.BaseUrl + loginEndpoint); err != nil {
-		return nil, errors.Wrap(err, "while performing HTTP GET on login endpoint")
+		return "", errors.Wrap(err, "while performing HTTP GET on login endpoint")
 	}
 
 	loginResp, err := p.httpClient.PostForm(p.config.DexConfig.BaseUrl+loginEndpoint, url.Values{
 		"login":    {p.config.UserCredentials.Username},
 		"password": {p.config.UserCredentials.Password},
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "while performing HTTP POST on login endpoint")
+		return "", errors.Wrap(err, "while performing HTTP POST on login endpoint")
 	}
 	defer closeRespBody(loginResp)
 	loginRespBody := readRespBody(loginResp)
 
 	if loginResp.StatusCode < 300 || loginResp.StatusCode > 399 {
-		return nil, fmt.Errorf("login - response error: '%s' - %s", loginResp.Status, loginRespBody)
+		return "", fmt.Errorf("login - response error: '%s' - %s", loginResp.Status, loginRespBody)
 	}
 
 	approvalEndpoint := loginResp.Header.Get("location")
 	if strings.Contains(approvalEndpoint, "#.*error") {
-		return nil, fmt.Errorf("approval - Redirected with error: '%s'", approvalEndpoint)
+		return "", fmt.Errorf("approval - Redirected with error: '%s'", approvalEndpoint)
 	}
+
+	return approvalEndpoint, nil
+}
+
+func (p *dexIdTokenProvider) tokenCall(approvalEndpoint string) (*url.URL, error) {
```

Comment:
Same suggestion as above.

what: receive token
how: performs a call to the approval endpoint

Reply:
Done.